airbyte-integrations/connectors/source-pinterest/source_pinterest/spec.json

```
@@ -0,0 +1,50 @@
+{
+  "documentationUrl": "https://docs.airbyte.io/integrations/sources/pinterest",
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Pinterest Spec",
+    "type": "object",
+    "required": ["client_id", "client_secret", "refresh_token", "start_date"],
+    "additionalProperties": false,
+    "properties": {
+      "client_id": {
+        "type": "string",
+        "title": "Client id",
+        "description": "Your Pinterest client id. See the <a href=\"https://developers.pinterest.com/docs/api/v5/#tag/Authentication\">docs</a> for instructions on how to generate it.",
+        "airbyte_secret": true
+      },
+      "client_secret": {
+        "type": "string",
+        "title": "Client secret",
+        "description": "Your Pinterest client secret. See the <a href=\"https://developers.pinterest.com/docs/api/v5/#tag/Authentication\">docs</a> for instructions on how to generate it.",
+        "airbyte_secret": true
+      },
+      "refresh_token": {
+        "type": "string",
+        "title": "Refresh token",
+        "description": "Your Pinterest refresh token. See the <a href=\"https://developers.pinterest.com/docs/api/v5/#tag/Authentication\">docs</a> for instructions on how to generate it.",
+        "airbyte_secret": true
+      },
+      "access_token": {
+        "type": "string",
+        "title": "Access token",
+        "description": "Your Pinterest access token. See the <a href=\"https://developers.pinterest.com/docs/api/v5/#tag/Authentication\">docs</a> for instructions on how to generate it.",
+        "airbyte_secret": true
+      },
+      "start_date": {
+        "type": "string",
+        "title": "Start date",
+        "description": "A date in the format YYYY-MM-DD.",
+        "examples": ["2020-06-01"]
+      },
+      "window_in_days": {
+        "type": "integer",
+        "title": "Window in days",
+        "description": "The amount of days for each data-chunk begining from start_date. Bigger the value - faster the fetch. (Min=1, as for a Day; Max=185, as for a Year).",
```

maybe silly question but 185 days does not equal a year, why does the comment say `Max=185 as for a Year`? 

```suggestion
        "description": "How many days to fetch data for at once, beginning from the Start Date. The bigger the value - the faster the fetch. Minimum value is 1, Maximum is 185.",
```

---

You are right. Silly typo from my side. Maximum range for dates is 185 day. I didn't find those limits in API's doc, but endpoint retuned error if you use date range more than 185. 

```
{
    "code": 1,
    "message": "The time range specified is too large, it must be less than 186 days"
}
```